### PR TITLE
➕ Install peft for prompt adapter support

### PR DESCRIPTION
**Pull request by prashantgupta24**
For allowing PEFT adapters to be loaded, we will need the `peft` library installed.

Upstream PR: https://github.com/vllm-project/vllm/pull/8123

### CODE REVIEW

**Comment by dtrifiro** [comment]
/lgtm

This pass after https://github.com/opendatahub-io/vllm/pull/126 is merged

File: Dockerfile.ubi [hunk 1]
```diff
@@ -180,6 +180,10 @@ RUN --mount=type=cache,target=/root/.cache/pip \
     --mount=type=cache,target=/root/.cache/uv \
     uv pip install https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.2/flashinfer-0.1.2+cu121torch2.4-cp311-cp311-linux_x86_64.whl
 
+# Install peft for prompt adapter support
+RUN --mount=type=cache,target=/root/.cache/pip \
+    pip install peft==0.12.0
```
Comment (dtrifiro) [inline Dockerfile.ubi:185]: Could you use `uv` here aswell?

**Comment by dtrifiro** [comment]
Thanks, forcing merge as smoke test job is currently failing due to a openshift CI outage on the GPU nodes cluster.

File: Dockerfile.ubi [hunk 2]
```diff
@@ -180,6 +180,10 @@ RUN --mount=type=cache,target=/root/.cache/pip \
     --mount=type=cache,target=/root/.cache/uv \
     uv pip install https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.2/flashinfer-0.1.2+cu121torch2.4-cp311-cp311-linux_x86_64.whl
 
+# Install peft for prompt adapter support
```
Comment (dtrifiro) [inline Dockerfile.ubi:183]: Sorry if I get back to this, but I just realized this is something which is only supported in the adapter, right? In that case this should be a dependency in `vllm-tgis-adapter` and not something installed here.

Comment (prashantgupta24) [reply Dockerfile.ubi:183]: This is something that is needed by vllm explicitly - https://github.com/opendatahub-io/vllm/blob/7a21f52109eb5102fc146b9025279ef4459d6435/vllm/prompt_adapter/models.py#L93

Comment (prashantgupta24) [reply Dockerfile.ubi:183]: Also see these lines:

Comment (maxdebayser) [reply Dockerfile.ubi:183]: I can confirm that this is needed to run the upstream unit tests.

**Comment by prashantgupta24** [comment]
Will try to get it merged in upstream https://github.com/vllm-project/vllm/pull/8123/files

**Pull request closed without merging**